api/v1beta1/grafana_types.go

```diff
@@ -139,6 +141,9 @@ type GrafanaClient struct {
 	// +nullable
 	// If the operator should send it's request through the grafana instances ingress object instead of through the service.
 	PreferIngress *bool `json:"preferIngress,omitempty"`
+	// +nullable
+	// If the operator should send it's request through the grafana instances HTTPRoute object instead of through the service.
+	PreferHTTPRoute *bool `json:"preferHTTPRoute,omitempty"`
```

Comment:
It's up for a discussion amongst maintainers, but I feel like it's not worth implementing as it would require too much complexity and wider cluster permissions for properly detecting protocol (HTTP/HTTPS).